File: app/src/main/java/com/example/android/soundtechsensors/BatteryActivity.java
package com.example.android.soundtechsensors;

import android.annotation.SuppressLint;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.BatteryManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;
import android.widget.Toast;

public class BatteryActivity extends AppCompatActivity {

   // public Context context;
    TextView currentBattery;
    private Context context;
    private int mBatteryLevel;
    private IntentFilter ifilter;


    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.battery_sensor);

        currentBattery = findViewById(R.id.current_battery);
        int i = registerMyReceiver();
        currentBattery.setText((i) + " percent");
    }

    /*public int getBatteryPct(Context context) {
        IntentFilter ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
        Intent batteryStatus = context.registerReceiver(null, ifilter);

      //  int level = batteryStatus.getIntExtra(EXTRA_LEVEL, -1);
     //   int scale = batteryStatus.getIntExtra(EXTRA_SCALE, -1);
        assert batteryStatus != null;
        String temp = batteryStatus.getStringExtra(EXTRA_TEMPERATURE);

       // return (int) (level / (float)scale);
        return Integer.parseInt(temp);
    }*/

/*    BroadcastReceiver mBatteryReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            mBatteryLevel = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
            Toast.makeText(context, "Battery" + mBatteryLevel, Toast.LENGTH_LONG).show();

        }
    };*/

/*    private String registerMyReceiver() {
        mBatteryLevelFilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
        registerReceiver(mBatteryReceiver, mBatteryLevelFilter);
        return new String(String.valueOf(mBatteryLevel));
    }*/

    BroadcastReceiver mBatteryReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            int level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
            int scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, 100);
            int percent = (level * 100) / scale;
            float batteryPct = level / (float)scale;

           // return batteryPct;

            Toast.makeText(context, "Current Battery Level: " + level, Toast.LENGTH_LONG).show();
        }
    };

    private int registerMyReceiver() {
        ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
        registerReceiver(mBatteryReceiver, ifilter);
       // return (BatteryManager.EXTRA_LEVEL);

       // return mBatteryLevel;
        return 0;
    }
}

File: app/src/main/java/com/example/android/soundtechsensors/BatteryActivity.kt
package com.example.android.soundtechsensors

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.BatteryManager
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.animation.AlphaAnimation
import android.widget.TextView
import android.widget.Toast
import com.google.firebase.analytics.FirebaseAnalytics
import kotlinx.android.synthetic.main.battery_sensor.*

class BatteryActivity : AppCompatActivity() {

    internal lateinit var battery_text: TextView
    internal lateinit var currentBattery: TextView
    private val context: Context? = null
    private val mBatteryLevel: Int = 0
    private var ifilter: IntentFilter? = null
   // private val level = registerMyReceiver()

    // Initiate Firebase Analytics
    private var mFirebaseAnalytics: FirebaseAnalytics? = null

    //Initiate Broadcast Receiver to utilize BatteryManager Features
    //Is parsed by registerMyReceiver() to the textview
    private var mBatteryReceiver: BroadcastReceiver = object : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            val level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
            val scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, 100)
            val percent = level * 100 / scale
            val batteryPct = level / scale.toFloat()

            currentBattery.text = "$level"

           // Toast.makeText(context, "Current Battery Level: $level", Toast.LENGTH_LONG).show()
        }
    }

    ////////////////////////////////////////////////////////////////////////////////////////////////

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.battery_sensor)

        battery_text = findViewById(R.id.battery) as TextView
        currentBattery = findViewById(R.id.current_battery) as TextView

        val `in` = AlphaAnimation(0.0f, 1.0f)
        `in`.duration = 1500
        battery_text.startAnimation(`in`)
        currentBattery.startAnimation(`in`)

        // Obtain the FirebaseAnalytics instance.
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(this)

        currentBattery = findViewById(R.id.current_battery)
        currentBattery.text = "${registerMyReceiver()} percent"

    }

    ////////////////////////////////////////////////////////////////////////////////////////////////

    private fun registerMyReceiver(): BroadcastReceiver {
        ifilter = IntentFilter(Intent.ACTION_BATTERY_CHANGED)
        registerReceiver(mBatteryReceiver, ifilter)

        return mBatteryReceiver
    }

    //This will add functionality to the menu button within the action bar
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.navigation_menu, menu)
        return true
    }
}